src/vs/workbench/browser/parts/views/viewsService.ts

```diff
@@ -239,8 +249,10 @@ export class ViewsService extends Disposable implements IViewsService {
 	}
 
 	getFocusedViewName(): string {
-		const viewId: string = this.contextKeyService.getContextKeyValue(FocusedViewContext.key) ?? '';
-		return this.viewDescriptorService.getViewDescriptorById(viewId.toString())?.name ?? '';
+		const key = this.contextKeyService.getContextKeyValue(FocusedViewContext.key);
+		const viewTitle: string | undefined = key ? this.viewDescriptorService.getViewDescriptorById(key.toString())?.name : undefined;
+		const containerTitle: string | undefined = !viewTitle ? this._openViewContainerTitle : undefined;
+		return viewTitle ?? containerTitle ?? '';
```

Comment:
Should we check if view container is in focus instead of visibility? Because visibility is true even when focus is in editor.

Reply:
Yeah, is there a way to do that currently? @sandy081

Reply:
We should introduce a new method in view pane container to know if a view pane container is focused or not. Probably we should the same for view instead of depending on the context key. I feel it weird to use context key to know the focused view.

Reply:
we could save it in a variable vs a context key, but I think it would add work to have a `focused` property on each view. then, we'd have to iterate through the views to look for one with `focused:true`

Reply:
also, I just reread your above comment. when I click in the editor, there is not a focused view - as expected.

Reply:
but we know when a view pane container is focused based on this event, so seems unnecessary imo to add a method. maybe i'm misunderstanding what you're suggesting.

Reply:
also, when I click in the editor, the view is set to none, so pretty sure this works.